### PR TITLE
Amazon Q : Fixing the inappropriate yellow lines shown in the editor file using CodeSnippet.

### PR DESCRIPTION
**Pull request by laileni-aws**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Problem
- If user triggers a scan and modifies the code in the editor, the security issue shows yellow squiggle lines at different place in the editor.
image

## Solution:
- To avoid this we compare the existing code diff from editor with the codeSnippet from response. If both matches we show the issue to the user if not we drop the suggestion.
- Implemented only for Auto-Scans.

## Checklist
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 1]
```diff
@@ -375,50 +373,69 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
...
+            if (file?.isDirectory == false) {
+                runReadAction {
+                    FileDocumentManager.getInstance().getDocument(file)
+                }?.let { document ->
+
+                    val documentLines = document.getText().split("\n")
+                    val (startLine, endLine) = recommendation.run { startLine to endLine }
+                    var shouldShow = true
+
+                    for (codeSni in recommendation.codeSnippet) {
```
Comment (ashishrp-aws) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:399]: nit: can we call it codeSni something else...like codeText or codeLine??

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 2]
```diff
@@ -375,50 +373,69 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
...
+                        if (codeSni.number in startLine..endLine) {
+                            val documentLine = documentLines.getOrNull(lineNumber)
+                            if (documentLine != codeSni.content) {
+                                shouldShow = false
+                                break
+                            }
+                        }
+                    }
+
+                    if (shouldShow) {
```
Comment (ashishrp-aws) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:410]: nit: shouldShow is vague name. change it to shouldDisplayIssue or something else.

**Review by ashishrp-aws — APPROVED** [review]

**Comment by manodnyab** [comment]
Can you please add a screenshot of the error?

**Comment by laileni-aws** [comment]
> Can you please add a screenshot of the error?

Added screenshot, If user trigger a File scan with a text change and while we get back the response if user does the other change we will show the issue. Expected behavior is to drop the issue and trigger a new file scan which is implemented in this PR.

**Review by manodnyab — APPROVED** [review]

**Pull request merged**